doc/code-tests.md

```diff
@@ -0,0 +1,190 @@
+# Code tests
+
+For its front end, Numberscope uses [`vitest`](https://vitest.dev/) for unit
+tests and low-level integration tests that run without a browser, and
+[`Playwright`](https://playwright.dev/) for higher-level integration and
+end-to-end integration tests that run in a browser
+
+### Running tests
+
+The package manager is set up with scripts that make basic running of all
+tests very convenient. To run all unit (vitest) tests, execute
+`npm run test:unit`. To run all end-to-end (Playwright) tests, use
+`npm run test:e2e`.
+
+A couple of additional points about the end-to-end tests:
+
```

Comment:
I added a paragraph to this effect (I think I got the gist of what you wanted). Please take a look.

Reply:
Also, another comment, it might be nice to include very briefly in the documentation something about what test:e2e reports at the end.  Specifically, at the end I see:
```
  "status": "passed",
  "status": "passed",
echo 'All tests passed' > e2e/certificate
```
I take it that the reason there are two status:passed lines is because there were two batches of tests?  If so, it's helpful to know these are the expected lines in the case of success.

Reply:
> I was just trying this out again and now when I run `npm run test:e2e` twice after a fresh pull([14b24f6](https://github.com/numberscope/frontscope/commit/14b24f680ac643aebae8495721270282e7d51595)..[31bbe08](https://github.com/numberscope/frontscope/commit/31bbe0819dff8979fec8defb91bff09415b435ac)), it will test everything both times (i.e. not report that tests already succeeded and decline to run the second time).

Hmm, I tried it on my end and it said `make: 'e2e/certificate' is up to date.` for me. Please when you pull the latest change, try again, two `npm run test:e2e` commands in immediate succession, and if the second doesn't just say the certificate is up to date, please post the full output of both invocations, thanks.

Reply:
> it might be nice to include very briefly in the documentation something about what test:e2e reports at the end.

Hopefully OK in the latest commit?

Reply:
the doc changes look good.  resolving.